django/db/backends/base/creation.py

```diff
@@ -268,6 +272,13 @@ def destroy_test_db(self, old_database_name=None, verbosity=1, keepdb=False, num
             settings.DATABASES[self.connection.alias]["NAME"] = old_database_name
             self.connection.settings_dict["NAME"] = old_database_name
 
+    def _restore_initial_data_migrations(self):
+        # Load initial data migrations, if any.
+        if hasattr(connections[self.connection.alias], "_test_serialized_contents"):
+            connections[self.connection.alias].creation.deserialize_db_from_string(
+                connections[self.connection.alias]._test_serialized_contents
+            )
```

Comment:
Something like: `You are using --keepdb option with database SERIALIZE option to False. If you have data migrations and using TransactionTestCase, be aware that, by preventing initial serialisation, the database at the end of the test will be empty.`
Sounds not so great :'(

Another solution would be to enforce `SERIALIZE` to `True` in case of `--keepdb` option (and warn the user about this behaviour).